File: QuickSort/QuickSort.py
#!/usr/local/bin/python

def swap(array, i, j):
	tmp = array[j]
	array[j] = array[i]
	array[i] = tmp
	
def Partition(array, lo, hi):
	pivot = array[hi]
	i = lo-1
	for j in range(lo,hi+1):
		if array[j] <= pivot:
			i = i+1
			if i != j:
				swap(array,i,j)
	return i
				
def QuickSort(array, lo, hi):
	if lo < hi:
		p = Partition(array, lo, hi)
		QuickSort(array, lo, p-1)
		QuickSort(array, p+1, hi)
	
s = input()
inputArray = [int(x) for x in s.split()]
QuickSort(inputArray, 0, len(inputArray)-1)
print(" ".join(str(x) for x in inputArray))

File: QuickSort/QuickSort.pl
#!/usr/bin/perl

sub swap {
	my $tmp = $inputArray[$_[1]];
	$inputArray[$_[1]] = $inputArray[$_[0]];
	$inputArray[$_[0]] = $tmp;
}

sub Partition {
	my $pivot = $inputArray[$_[1]];
	my $i = $_[0]-1;
	for (my $j = $_[0]; $j <= $_[1]; $j++) {
		if ($inputArray[$j] <= $pivot) {
			$i ++;
			if ($i != $j) {
				swap($i,$j);
			}
		}
	}
	return $i;
}

sub QuickSort {
	if ($_[0] < $_[1]) {
		$p = Partition($_[0],$_[1]);
		QuickSort($_[0],$p-1);
		QuickSort($p+1,$_[1]);
	}
}

my $str = <STDIN>;
chomp $str;
@inputArray = split(' ', $str);
QuickSort(0,@inputArray-1);
print("@inputArray\n");